### PR TITLE
Processing module for ExfilDetect.com added.

### PR DESCRIPTION
**Pull request by 0x71**
Hi,

I just added a new processing module allowing automated pcap scans for advanced data exfiltration using a new platform called [ExfilDetect](https://exfildetect.com). ExfilDetect is a free online service that analyzes pcap files allowing the identification of hidden data exfiltration.
 
To be more precise, it detects advanced steganographic data exfiltration and protocol tunneling hidden in DNS and ICMP (more protocols are to be added in the future). The automated detection is based on a novel approach investigated during my Masters thesis that will be published in March 2017. The new approach uses behavior-based statistical analysis and machine learning. 

If you are interested in how an analysis looks like, take a look [here](https://exfildetect.com/index.php?page=analyses&query=326932326b39b0bd8188c4fade9ca12b20c6f81075613dfbd7a1bfd900cb42f6).

Best regards,
Nils

### CODE REVIEW

File: conf/processing.conf [hunk 1]
```diff
@@ -35,6 +35,13 @@ enabled = yes
 [dumptls]
 enabled = yes
 
+[exfildetect]
+enabled = yes
```
Comment (doomedraven) [inline conf/processing.conf]: set no by default ;)

**Comment by 0x71** [comment]
@doomedraven Thanks ;-)

**Comment by jbremer** [comment]
Although your project looks interesting, we can't integrate it. Currently we don't have integration with any 3rd party services enabled by default (there's just VirusTotal and I disabled it by default just now). This is due to the sensitivity of a lot of our users their Cuckoo instances. I welcome you to implement your ex-filtration analysis ideas in our own `.pcap` reading processing module and we can work from there. See also https://github.com/cuckoosandbox/cuckoo/blob/package/cuckoo/processing/network.py

**Comment by 0x71** [comment]
@jbremer Thanks for your feedback! I am aware of the sensitive data that might be generated by some private Cuckoo instances. That is why the source code of my project (the pcap analysis and machine learning part) is to be published under GNU GPLv3. Like for your MISP integration, the users could use this code to run a local analysis instance of ExfilDetect by changing the specified address in conf/processing.conf. Is that a possibility for you? Otherwise, I will consider the integration in your `.pcap` reading module (which is probably not that simple).

**Comment by jbremer** [comment]
Is your library doing any performance / resource needy stuff? E.g., MISP requires a MySQL server, its own API endpoint, etc. If your code could be used in Cuckoo simply by making a Python package out of it and importing that (such as we do with, e.g., my https://github.com/jbremer/httpreplay library), that'd be a much, much easier to setup, manage & maintain add-on as compared to an external service.

**Comment by 0x71** [comment]
A Python package sounds good. The framework needs some performance to create the initial baseline (optional) that is optimized for a local network. However, a general baseline that was created during my Masters thesis could be shipped with the package. Currently, the framework requires a MySQL Server to save the baseline, analysis results, and to visualize the results using a web interface. For the Cuckoo integration, I could build a "light version" that can perform pcap analyses without the MySQL Server. Of course, I could leave out the web interface, too. In consequence, this version could basically run with two local python instances (pcap sniffer/analyzer + classification server).

**Comment by jbremer** [comment]
@0x71 Ok, how do you suggest we move forward? Shall I close this issue and you make a new PR once you've got the whole Python package + classification server etc going on?

**Comment by 0x71** [comment]
@jbremer That is a good proposal. I will make a new PR for the whole python package.

**Comment by jbremer** [comment]
Ok, closing this one then. Looking forward to it!

**Pull request closed without merging**